src/ChunkingContext.jl

```diff
@@ -0,0 +1,119 @@
+# When splitting the work among multiple tasks, we aim for each task to have at least this
+# many bytes of input to work on (even if it means that some tasks will have nothing to process)
+# This is to avoid overhead from too many task switches.
+# TODO: make this configurable and find a good default (the current 16 KiB is a guess)
```

Comment:
Please can you go through the `TODO` comments in this package and (i) open an issue for each of them, (ii) link to that issue in the TODO comment?